eppo-sdk-test/store/ConfigurationStoreTest.cs

```diff
@@ -5,37 +5,77 @@
-        var banditFlags = new BanditFlags();
-        var response = new FlagConfigurationResponse() {
-            Bandits = banditFlags,
+
+    private ConfigurationStore? Store;
+    MemoryCache? ConfigCache;
+    MemoryCache? ModleCache;
+    MemoryCache? BanditFlagCache;
+
+    BanditFlags BanditFlags;
+    
+
+    [SetUp]
+    public void TestSetUp() {
+        ConfigCache = new CacheHelper(Constants.MAX_CACHE_ENTRIES).Cache;
+        ModleCache = new CacheHelper(Constants.MAX_CACHE_ENTRIES).Cache;
+        BanditFlagCache = new CacheHelper(Constants.MAX_CACHE_ENTRIES).Cache;
+        BanditFlags = new BanditFlags();
+        var response = new FlagConfigurationResponse()
+        {
+            Bandits = BanditFlags,
             Flags = new Dictionary<string, Flag>()
         };
-        var banditResponse = new BanditModelResponse() {
+        var banditResponse = new BanditModelResponse()
+        {
             Bandits = new Dictionary<string, Bandit>()
         };
 
         var mockRequester = new Mock<IConfigurationRequester>();
-        mockRequester.Setup(m=>m.FetchFlagConfiguration()).Returns(response);
-        mockRequester.Setup(m=>m.FetchBanditModels()).Returns(banditResponse);
+        mockRequester.Setup(m => m.FetchFlagConfiguration()).Returns(response);
+        mockRequester.Setup(m => m.FetchBanditModels()).Returns(banditResponse);
+
+        Store = new ConfigurationStore(mockRequester.Object, ConfigCache, ModleCache, BanditFlagCache);
+    }
+    [Test]
+    public void ShouldStoreAndGetBanditFlags()
+    {
+
+        Store!.FetchConfiguration();
+
+        Assert.That(Store.GetBanditFlags(), Is.EqualTo(BanditFlags));
+    }
+
+    [Test]
+    public void ShouldClearCacheOnFetch()
+    {
+        // MemoryCache.Clear is non-overridable so we can't mock it and verify the method call
+        // Instead, we populate the caches and ensure they're empty when they should be.
+        ConfigCache!.Set<string>("foo", "bar", new MemoryCacheEntryOptions().SetSize(1));
+        ModleCache!.Set<string>("foo", "bar", new MemoryCacheEntryOptions().SetSize(1));
+        BanditFlagCache!.Set<string>("foo", "bar", new MemoryCacheEntryOptions().SetSize(1));
+
+        Multiple(() =>
+        {
+            That(ConfigCache, Has.Count.EqualTo(1));
+            That(ModleCache, Has.Count.EqualTo(1));
+            That(BanditFlagCache, Has.Count.EqualTo(1));
+        });
 
-        var store = new ConfigurationStore(mockRequester.Object, configCache, modelCache, banditFlagCache);
-        store.FetchConfiguration();
+        Store!.FetchConfiguration();
 
-        Assert.That(store.GetBanditFlags(), Is.EqualTo(banditFlags));
+        Multiple(() =>
+        {
+            That(ConfigCache, Has.Count.EqualTo(0));
+            That(ModleCache, Has.Count.EqualTo(0));
+            That(BanditFlagCache, Has.Count.EqualTo(1));
+            That(BanditFlagCache.Get<BanditFlags>("bandit_flags"), Is.EqualTo(BanditFlags));
+        });
```

Comment:
Do these tests know too much about the implementation? At the top level we are interested in fixing the issue of:

* flags in cache
* API removes a flag
* that flag is still in cache

This is how I structured the test for ruby (https://github.com/Eppo-exp/ruby-sdk/pull/18/files#diff-b45b9ee73f663c0fc0e6959ed7e5a7433c48712e6edbfd21aa6a12e97f7ccb72) do you think both approaches make sure this is not reproduced?

Reply:
I like your Ruby test - much more succinct and readable. Exposed a shortcut method in `ConfigurationStore` to make these tests cleaner.